Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -314,11 +318,93 @@ void CookieStore::remove(CookieStoreDeleteOptions&& options, Ref<DeferredPromise
     set(WTFMove(initOptions), WTFMove(promise));
 }
 
+void CookieStore::cookiesAdded(const String& host, const Vector<Cookie>& cookies)
+{
+    UNUSED_PARAM(host);
+    if (!m_hasChangeEventListener)
+        return;
+
+    auto* context = scriptExecutionContext();
+    if (!context)
+        return;
+
+    auto& document = *downcast<Document>(context);
+    UNUSED_PARAM(document);
+    ASSERT(host == document.url().host().toString());
+
+    Vector<CookieListItem> changedCookieList;
+    Vector<CookieListItem> deletedCookieList;
+    auto currentTime = WallTime::now().secondsSinceEpoch().milliseconds();
+    for (auto cookie : cookies) {
+        if (cookie.expires && *cookie.expires < currentTime) {
+            cookie.value = nullString();
+            deletedCookieList.append(CookieListItem { cookie });
```

Comment:
It you create eventInit before the `for` loop then you can append directly to `eventInit.deleted` instead of having a temporary vector.